nemo/collections/speechlm2/parts/metrics/asr_bleu.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import defaultdict
+
+import sacrebleu
+import torch
+from whisper_normalizer.english import EnglishTextNormalizer
+
+from nemo.collections.asr.models import ASRModel
+from nemo.collections.common.parts.optional_cuda_graphs import WithOptionalCudaGraphs
+from nemo.collections.speechlm2.parts.precision import fp32_precision
+from nemo.collections.speechlm2.parts.pretrained import load_pretrained_nemo
+from nemo.utils import logging
+
+
+class ASRBLEU:
```

Comment:
Curious: what is the use of this?

Reply:
It measures the similarity between generated model speech responses and the reference text using a pretrained ASR.